### PR TITLE
Reorder publications on details of deal.II from most recent to oldest

### PR DESCRIPTION
**Pull request by luca-heltai**
By looking at the publication list, I noticed that in the details section we had no logical nor temporal order in place. I reordered the manually generated list of publications on details of deal.II from most recent to oldest, and added details on a publication that was accepted on ACM.

### CODE REVIEW

**Comment by luca-heltai** [comment]
better reviewed without white space changes.

https://github.com/dealii/publication-list/pull/362/files?w=1

**Review by marcfehling — APPROVED** [review]

**Comment by marcfehling** [comment]
> better reviewed without white space changes.
> 
> https://github.com/dealii/publication-list/pull/362/files?w=1

That is a need feature!

/rebuild

File: publications.include [hunk 1]
```diff
@@ -92,333 +90,326 @@
...
+        <a href="https://arxiv.org/abs/2206.06512">arXiv:2206.06512</a>, 2022.
+      </li>
+
+      <li>
+        L. Heltai, W. Bangerth, M. Kronbichler, A. Mola
+        <br />
+        <strong>Propagating geometry information to finite element computations
+        </strong>
+        <br />
+        ACM Transactions on Mathematical Software, vol. 47, article 4,
```
Comment (kronbichler) [inline publications.include:257]: I think this needs to be
```suggestion
        ACM Transactions on Mathematical Software, vol. 47, issue 4, article 32,
```

Comment (kronbichler) [reply publications.include:257]: I realize we write just the equivalent of `vol. 47, 4` below.

**Pull request merged**